homeassistant/components/yalexs_ble/config_flow.py

```diff
@@ -0,0 +1,245 @@
+            await self.async_set_unique_id(
+                discovery_info.address, raise_on_progress=False
+            )
+            self._abort_if_unique_id_configured()
+            try:
+                await validate_lock(local_name, discovery_info.device, key, slot)
+            except InvalidKeyFormat:
+                errors[CONF_KEY] = "invalid_key_format"
+            except InvalidKeyIndex:
+                errors[CONF_SLOT] = "invalid_key_index"
+            except (DisconnectedError, AuthError, ValueError):
+                errors[CONF_KEY] = "invalid_auth"
+            except BleakError:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected error")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(
+                    title=local_name,
+                    data={
+                        CONF_LOCAL_NAME: discovery_info.name,
+                        CONF_ADDRESS: discovery_info.address,
+                        CONF_KEY: key,
+                        CONF_SLOT: slot,
+                    },
+                )
+
+        if discovery := self._discovery_info:
+            self._discovered_devices[discovery.address] = discovery
+        else:
+            current_addresses = self._async_current_ids()
+            current_unique_names = {
+                entry.data.get(CONF_LOCAL_NAME)
+                for entry in self._async_current_entries()
+                if local_name_is_unique(entry.data.get(CONF_LOCAL_NAME))
+            }
+            for discovery in async_discovered_service_info(self.hass):
+                if (
+                    discovery.address in current_addresses
+                    or discovery.name in current_unique_names
+                    or discovery.address in self._discovered_devices
+                    or YALE_MFR_ID not in discovery.manufacturer_data
+                ):
+                    continue
+                self._discovered_devices[discovery.address] = discovery
+
+        if not self._discovered_devices:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_ADDRESS): vol.In(
+                    {
+                        service_info.address: f"{service_info.name} ({service_info.address})"
+                        for service_info in self._discovered_devices.values()
+                    }
+                ),
+                vol.Required(CONF_KEY): str,
+                vol.Required(CONF_SLOT): int,
+            }
+        )
+        integration = await async_get_integration(self.hass, DOMAIN)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors,
+            description_placeholders={"docs_url": integration.documentation},
```

Comment:
We've been removing links to integration docs as we've added a link to the integration docs to all config flows in the GUI under the help icon.

Reply:
Will drop this and tell them to check the documentation instead

Reply:
https://github.com/home-assistant/core/pull/76691